src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -232,13 +232,13 @@ public function searchBarAction(Request $request)
             $limit =
             $token = null;
         } else {
-            $ip = $session->get('_profiler_search_ip');
-            $method = $session->get('_profiler_search_method');
-            $url = $session->get('_profiler_search_url');
-            $start = $session->get('_profiler_search_start');
-            $end = $session->get('_profiler_search_end');
-            $limit = $session->get('_profiler_search_limit');
-            $token = $session->get('_profiler_search_token');
+            $ip = $request->get('ip', $session->get('_profiler_search_ip'));
```

Comment:
please use ``$request->query->get()``. There is no need to read it from the POST params or the attributes as it can only be in the query string (same for others).

And using the query string should also work when the session is disabled (a few lines above).
thus, we should avoid reading from the session if we have the query string param IMO instead of always reading the session value just to compute the default value for the next call (methods calls are not lazy in PHP)